### PR TITLE
Fix/pro 1470

### PR DESCRIPTION
**Pull request by taylorferran**
## Description
Changing Klima to use advanced routes api

Testing:
Works with key based to smart account
Works with smart account to smart account
Key based to key based currently fails but it's also failing on the dev build with transaction reverted.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/TransactionPreview/ActionPreview.tsx [hunk 1]
```diff
@@ -687,7 +687,7 @@ const ActionPreview = ({
                 {!!cost && (
                   <>
                     <Text size={12} marginLeft={8} color={theme.color?.text?.innerLabel} medium block>
-                      Gas price
+                      Gas price1
```
Comment (MayukhSamadder2000) [inline src/components/TransactionPreview/ActionPreview.tsx:690]: Revert this

File: src/utils/transaction.ts [hunk 2]
```diff
@@ -320,7 +334,33 @@ export const klimaDaoStaking = async (
         },
       };
     }
+/// XXX
```
Comment (MayukhSamadder2000) [inline src/utils/transaction.ts:337]: remove this comment

File: src/utils/transaction.ts [hunk 3]
```diff
@@ -448,8 +488,18 @@ export const buildCrossChainAction = async (
       if (fromChainId !== CHAIN_ID.POLYGON) {
         try {
           let destinationTxns: ICrossChainActionTransaction[] = [];
-          let transactions: ICrossChainActionTransaction[] = [];
+          const { items: advancedRouteSteps } = await sdk.getStepTransaction({ route:routeToUSDC });
 
+          // XXX
```
Comment (MayukhSamadder2000) [inline src/utils/transaction.ts:493]: remove this comment

File: src/utils/transaction.ts [hunk 4]
```diff
@@ -458,8 +508,32 @@ export const buildCrossChainAction = async (
               createTimestamp,
               status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
             },
-          ];
+          ]; */
```
Comment (MayukhSamadder2000) [inline src/utils/transaction.ts:511]: commented code isn't required I guess

File: src/utils/transaction.ts [hunk 5]
```diff
@@ -484,7 +558,7 @@ export const buildCrossChainAction = async (
             };
 
             transactions = [approvalTransaction, ...transactions];
-          }
+          } */
```
Comment (MayukhSamadder2000) [inline src/utils/transaction.ts:561]: same as above

File: src/utils/transaction.ts [hunk 6]
```diff
@@ -880,6 +954,8 @@ export const buildCrossChainAction = async (
         route,
       };
 
+
+      // XXX
```
Comment (MayukhSamadder2000) [inline src/utils/transaction.ts:958]: remove this

File: src/components/TransactionBlock/KlimaStakingTransactionBlock.tsx [hunk 7]
```diff
@@ -312,45 +335,43 @@ const KlimaStakingTransactionBlock = ({
...
-        } else {
-          resetRoutes();
-          setTransactionBlockFieldValidationError(
-            transactionBlockId,
-            'route',
-            'Please try with different inputs/amount'
-          );
-        }
       } catch (err) {
+        console.log(err);
```
Comment (MayukhSamadder2000) [inline src/components/TransactionBlock/KlimaStakingTransactionBlock.tsx:369]: remove this log

File: src/components/TransactionBlock/KlimaStakingTransactionBlock.tsx [hunk 8]
```diff
@@ -183,6 +188,19 @@ const KlimaStakingTransactionBlock = ({
...
+
+    routes.forEach((route) => {
+      const { gasCostUSD, fromAmountUSD } = route;
+      if (!gasCostUSD) return;
+      if (+fromAmountUSD - +gasCostUSD > minAmount) bestRoute = route;
+    });
+
+    return bestRoute;
+  };
+
```
Comment (MayukhSamadder2000) [inline src/components/TransactionBlock/KlimaStakingTransactionBlock.tsx:203]: We do have a similar function, do one thing make this function generic, as in move it to utils or something

**Comment by poocart** [comment]
Work has been done by https://linear.app/pillarproject/issue/PRO-1827/klima-website-wrapper

**Pull request closed without merging**